source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -257,9 +257,11 @@ void ShaderGraph::addColorTransformNode(ShaderInput* input, const ColorSpaceTran
         ShaderNode* colorTransformNode = colorTransformNodePtr.get();
         ShaderOutput* colorTransformNodeOutput = colorTransformNode->getOutput(0);
 
-        // TODO: For now copy the value of the input to the transform node. In the future we don't want to do things
-        // this way. Instead we want to set the color transform uniform to be equal to the input uniform.
-        colorTransformNode->getInput(0)->value = input->value;
+        ShaderInput* shaderInput = colorTransformNode->getInput(0);
+        shaderInput->variable = input->node->getName() + "_" + input->name;
+        shaderInput->value = input->value;
+        shaderInput->noRename = true;
+        _renameTransformMap[shaderInput->variable + "_cm_" + shaderInput->name] = shaderInput->variable;
```

Comment:
As noted below you don't need this. You can just copy over the variable you set above when you detect "non-renameable".